sklearn/linear_model/tests/test_logistic.py

```diff
@@ -121,16 +123,9 @@ def test_predict_3_classes():
     check_predictions(LogisticRegression(C=10), X_sp, Y2)
 
 
-def test_predict_iris():
-    # Test logistic regression with the iris dataset
-    n_samples, n_features = iris.data.shape
-
-    target = iris.target_names[iris.target]
-
-    # Test that both multinomial and OvR solvers handle
-    # multiclass data correctly and give good accuracy
-    # score (>0.95) for the training data.
-    for clf in [
+@pytest.mark.parametrize(
```

Comment:
On the top of my head, I recall that LBFGS does not converge with the 100 iterations on the iris. Do we want to add a `pytest.mark.filterwarnings` on the top of the test to avoid spurious `ConvergenceWarning`?

Reply:
This is unrelated to this PR, but yes: `pytest -x -Werror sklearn/linear_model/tests/test_logistic.py::test_predict_iris` gives `ConvergenceWarning: lbfgs failed to converge`. I'll add the filter.

Reply:
What is better: 1) Filter the ConvergenceWarning or 2) increase max_iter in lbfgs?

Reply:
1) Will remind us that there is still an issue with lbfgs;-) Done in https://github.com/scikit-learn/scikit-learn/pull/24767/commits/74058bedc29fdd407291ecd5cabc1316f3f7aaa8.

Reply:
And I opened #24819.